interview-questions.md

```diff
@@ -8,36 +8,56 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Your answer:
+Your answer: You can migrate the model from the Ruby console to add in the foreign key. The name of the foreign key would be an integer tied to a label so an example of this might look like - `student_id:`. This also explains why the foreign key would be connect to the Student model. 
 
-Researched answer:
+Researched answer: You can fix the mix and add in the foreign key by creating a migration to add the foreign keys using `add_foreign_key` before specifying the foreign key parameters. The name of the foreign key would need to describe or point towards the relationship between the two models used, since foreign keys link the two databases together and will output Boolean values, so in other words the foreign key name will always be `modelname_id` / cohort_id. The foreign key would be on the Student model because a student belongs to a cohort, so cohort_id would connect the Student model and Cohort model together. 
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: New, Show, Edit, and Destroy would be routes that always must be passed params because you wouldn't want these methods to be applied in mass but on a more case-by-case basis.  
 
-Researched answer:
+Researched answer: The RESTful routes that must always be passed params would be Edit, Update and Destroy (3 of 7 routes) because it must be specified individually as to what data needs to be modified or removed, otherwise without the params being there, these changes would effect the content in mass. 
 
 3. Name three rails generator commands. What is created by each?
 
-Your answer:
+Your answer: One command would be `rails s` activates rails API server so you can see the changes in real time and investigate errors that may come up as it gives decent information on that, another is `rails c` allows access to the rails console so you can view or create a model, and lastly `rails db:create` is what is used to create database migrations after the column content have been prior established. 
 
-Researched answer:
+Researched answer: First off, `rails g` or `rails generate` can be used to generate/create something new dependin upon what follows it, so `rails g controller` would create a new controller or `rails g resource...` would create a new resouce. Another option would be `rails g -h` is an important one because it gives more information about what commands are allow and the different synatx options for them, returning a list of general options. And finally, `rails destroy...` is useful for when you are beginning a project and may have excessive typos or wish change the relationship dynamic of the models; well this command will go through every path that a generator will use to create a file, only it will remove the fil instead. 
```

Comment:
Fantastic responses, as always!  Maybe work on some rails vocab to help your own answers.  You did great!